core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -302,6 +305,98 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
   }
 
   /**
+   * testSendBeforeAndAfterPartitionExpansion checks the partitioning behavior before and after partitions are added
+   *
+   * The specified partition-ids should be respected
+   */
+  @Test
+  def testSendBeforeAndAfterPartitionExpansion() {
+    val producer = createProducer(brokerList)
+
+    try {
+      // create topic
+      val leaders = TestUtils.createTopic(zkUtils, topic, 1, 2, servers)
+      val partition0 = 0
+
+      // make sure leaders exist
+      var leader0 = leaders(partition0)
+      assertTrue("Leader for topic \"topic\" partition 0 should exist", leader0.isDefined)
+
+      var responses0 =
+        for (i <- 1 to numRecords)
+        yield producer.send(new ProducerRecord[Array[Byte], Array[Byte]](topic, partition0, null, ("value" + i).getBytes))
```

Comment:
Can you try removing `[Array[Byte], Array[Byte]]` to see if it's inferred? If so, the same for other similar usages.